fuzz/src/feature_flags.rs

```diff
@@ -0,0 +1,95 @@
+// This file is Copyright its original authors, visible in version control
+// history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+use lightning::types::features::FeatureFlags;
+
+use crate::utils::test_logger;
+
+use std::ops::{Deref, DerefMut};
+
+/// Check various methods on [`FeatureFlags`] given `v` which should be equal to `feat` and an
+/// `old_v` which should be equal to `old_feat`.
+fn check_eq(v: &Vec<u8>, feat: &FeatureFlags, old_v: &mut Vec<u8>, old_feat: &mut FeatureFlags) {
+	assert_eq!(v.len(), feat.len());
+	assert_eq!(v.deref(), feat.deref());
+	assert_eq!(old_v.deref_mut(), old_feat.deref_mut());
+
+	let mut feat_clone = feat.clone();
+	assert!(feat_clone == *feat);
+
+	let mut feat_iter = feat.iter();
+	let mut vec_iter = v.iter();
+	assert_eq!(feat_iter.len(), vec_iter.len());
+	while let Some(feat) = feat_iter.next() {
+		let v = vec_iter.next().unwrap();
+		assert_eq!(*feat, *v);
+	}
+	assert!(vec_iter.next().is_none());
+
+	let mut feat_iter = feat_clone.iter_mut();
+	let mut vec_iter = v.iter();
+	assert_eq!(feat_iter.len(), vec_iter.len());
+	while let Some(feat) = feat_iter.next() {
+		let v = vec_iter.next().unwrap();
+		assert_eq!(*feat, *v);
+	}
+	assert!(vec_iter.next().is_none());
```

Comment:
Can you comment each of these blocks since they look almost the same, to highlight what is unique in the second case?